README.md

```diff
@@ -1,2 +1,555 @@
-# cli
-Clean and elegant CLI development kit.
+# {Name}
+
+[![Build](https://github.com/candy-kingdom/cli/workflows/Build/badge.svg)](https://github.com/JoshuaLight/chalice-restful/actions)
+
+_Clean and elegant CLI development kit._
+
+## Overview
+
+{Name} is a microframework that allows building declarative and nice-looking CLI apps.
+
+Consider the following simple program as a demonstration of {Name} capabilities:
+```py
+# say.py
+
+from {pyname} import cli, Arg, Flag, Opt
+
+@cli
+def say(phrase: Arg('a phrase to print'),
+        caps:   Flag('whether to print phrase in upper-case'),
+        times:  Opt[int]('how many times to print') = 1):
+    """Prints a phrase specified number of times."""
+
+    for i in range(0, times):
+        print(phrase.upper() if caps else phrase)
+
+if __name__ == '__main__':
+    cli()
+```
+
+It then can be used as follows:
+```
+$ python3 say.py "Hi!" 
+Hi!
+$ python3 say.py "Hi!" --caps
+HI!
+$ python3 say.py "Hi!" --times 3
+Hi!
+Hi!
+Hi!
+$ python3 say.py --help
+usage: say [-h] [--caps] [--times TIMES] phrase
+
+Prints a phrase specified number of times.
+
+positional arguments:
+  phrase         a phrase to print
+
+optional arguments:
+  -h, --help     show this help message and exit
+  --caps         whether to print phrase in upper-case
+  --times TIMES  how many times to print
+```
+
+## Install
+
+```
+$ pip install {pipname}
+```
+
+## Getting started
+
+_Note: the examples are not [PEP 8](https://www.python.org/dev/peps/pep-0008/#blank-lines) compatible: one blank line is used instead of two to separate top-level definitions._
+
+In general, writing a CLI app is very similar to writing a regular function.
+{Name} is based on this metaphor, allowing to describe the whole interface of the app using only a function signature.
+Consider the following example:
+```py
+# print.py
+
+from {pyname} import cli, Arg
+
+@cli
+def entry(phrase: Arg):
+    print(phrase)
+
+if __name__ == '__main__':
+    cli()
+```
+
+The script can be executed as a command line app:
+```
+$ python3 print.py "Hello, world!"
+Hello, world!
+```
+
+The main idea is very simple: you use the `cli` decorator to wrap a function that acts as an entry point of the application (`entry`), and then use the `cli()` to make things running.
+`Arg` is used to annotate positional arguments of the CLI.
+
+In the following section we'll discuss more deeply how to implement different types of parameters in {Name}.
+
+### Parameters
+
+There are three different type annotations that could be used to represent positional arguments, flags and options.
```

Comment:
I think it is simply _annotations_, without _type_.

Reply:
Do you mean it is not valid to say "type annotation" or it would be just a bit shorter to say "annotation"?

Reply:
The feature itself is called _annotations_. I believe that people sometimes call it _type annotations_ just because it is directly connected to _type hints_ (this is what annotations were designed for in the first place). However, in our case, we don't use annotations for type hinting, and this is why I think we should omit the word _type_.

Reply:
Agreed. Fixed!